src/main/java/net/coderbot/iris/config/IrisConfig.java

```diff
@@ -0,0 +1,125 @@
+    /**
+     * Represents if the current loaded shaderpack is in internal one or not
+     */
+    private boolean isInternal = false;
+
+    public IrisConfig() {
+        propertiesPath = FabricLoader.getInstance().getConfigDir().resolve("iris.properties");
+        try {
+            Files.createDirectories(FabricLoader.getInstance().getGameDir().resolve("shaderpacks"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * This sets the default shaderpack name for the shader properties to set.
+     * This value will only be used if the iris.properties file is not found
+     * @param shaderPackName the file name of the pack
+     * @return the shaderproperties with the set value
+     */
+    public IrisConfig setDefaultPack(String shaderPackName){
+        this.shaderpath = FabricLoader.getInstance().getGameDir().resolve("shaderpacks").resolve(shaderPackName);
+        return this;
+    }
+
+    /**
+     * Deserializes and serializes the config
+     * @throws IOException file exceptions
+     */
+    public void createAndLoadProperties() throws IOException {
+        deserialize();
+        serialize();
+    }
+
+    /**
+     * returns the path of the current shaderpack
+     * @return the path of the shaderpack
+     */
+    public Path getShaderPackPath(){
+        return shaderpath;
+    }
+
+    /**
+     * The path of the config file
+     * @return the path to config file
+     */
+    public Path getPropertiesPath(){
+        return propertiesPath;
+    }
+
+    /**
+     * returns whether or not the current shaderpack is internal
+     * @return if the shaderpack is internal
+     */
+
+    public boolean isInternal(){
+        return isInternal;
+    }
+
+    /**
+     * Returns the name of the shaderpack
+     * @return shaderpack name. If internal it returns "internal"
+     */
+
+    public String getShaderPackName(){
+        if (isInternal){
+            return "internal";
+        }
+        return shaderpath.getFileName().toString();
+    }
+
+    /**
+     * loads the config file and then populates the string, int, and boolean entries with the parsed entries
+     * @throws IOException if the file cannot be loaded
+     */
+
+    public void deserialize() throws IOException {
+        if (!propertiesPath.toFile().exists()){
+            return;
+        }
+        Properties properties = new Properties();
+        properties.load(Files.newInputStream(propertiesPath));
+        properties.forEach((key, value) -> {
+           if (key.toString().equals("shaderpack")){
+                this.shaderpath = Paths.get(value.toString());
+            }
+        });
+    }
+
+    /**
+     * Serializes the config into a file. Should be called after refreshing the values of entries
+     * @throws IOException file exceptions
+     */
+
+    public void serialize() throws IOException {
+        Properties properties = new Properties();
+        if (!shaderpath.toFile().exists()){
```

Comment:
Avoid toFile whenever possible! Use the correct method in [`Files`](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html) instead. As far as I know, attempting to convert a valid Path to a File when the Path is ie. in a custom filesystem (like inside a jar file) will crash.

In this case, use `File#exists`.